pkg/virt-launcher/virtwrap/converter/compute/clock.go

```diff
@@ -0,0 +1,122 @@
+	"strconv"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/watch/topology"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type ClockDomainConfigurator struct{}
+
+func (c ClockDomainConfigurator) Configure(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
+	if vmi.Spec.Domain.Clock != nil {
+		clock := vmi.Spec.Domain.Clock
+		newClock := &api.Clock{}
+		err := convert_v1_Clock_To_api_Clock(clock, newClock)
+		if err != nil {
+			return err
+		}
+		domain.Spec.Clock = newClock
+	}
+
+	// Make use of the tsc frequency topology hint
+	if topology.IsManualTSCFrequencyRequired(vmi) {
+		freq := *vmi.Status.TopologyHints.TSCFrequency
+		clock := domain.Spec.Clock
+		if clock == nil {
+			clock = &api.Clock{}
+		}
+		clock.Timer = append(clock.Timer, api.Timer{Name: "tsc", Frequency: strconv.FormatInt(freq, 10)})
+		domain.Spec.Clock = clock
+	}
+
+	return nil
+}
+
+func convert_v1_Clock_To_api_Clock(source *v1.Clock, clock *api.Clock) error {
+	if source.UTC != nil {
+		clock.Offset = "utc"
+		if source.UTC.OffsetSeconds != nil {
+			clock.Adjustment = strconv.Itoa(*source.UTC.OffsetSeconds)
+		} else {
+			clock.Adjustment = "reset"
+		}
+	} else if source.Timezone != nil {
+		clock.Offset = "timezone"
+		clock.Timezone = string(*source.Timezone)
+	}
+
+	if source.Timer != nil {
+		if source.Timer.RTC != nil {
+			newTimer := api.Timer{Name: "rtc"}
+			newTimer.Track = string(source.Timer.RTC.Track)
+			newTimer.TickPolicy = string(source.Timer.RTC.TickPolicy)
+			newTimer.Present = boolToYesNo(source.Timer.RTC.Enabled, true)
+			clock.Timer = append(clock.Timer, newTimer)
+		}
+		if source.Timer.PIT != nil {
+			newTimer := api.Timer{Name: "pit"}
+			newTimer.Present = boolToYesNo(source.Timer.PIT.Enabled, true)
+			newTimer.TickPolicy = string(source.Timer.PIT.TickPolicy)
+			clock.Timer = append(clock.Timer, newTimer)
+		}
+		if source.Timer.KVM != nil {
+			newTimer := api.Timer{Name: "kvmclock"}
+			newTimer.Present = boolToYesNo(source.Timer.KVM.Enabled, true)
+			clock.Timer = append(clock.Timer, newTimer)
+		}
+		if source.Timer.HPET != nil {
+			newTimer := api.Timer{Name: "hpet"}
+			newTimer.Present = boolToYesNo(source.Timer.HPET.Enabled, true)
+			newTimer.TickPolicy = string(source.Timer.HPET.TickPolicy)
+			clock.Timer = append(clock.Timer, newTimer)
+		}
+		if source.Timer.Hyperv != nil {
+			newTimer := api.Timer{Name: "hypervclock"}
+			newTimer.Present = boolToYesNo(source.Timer.Hyperv.Enabled, true)
+			clock.Timer = append(clock.Timer, newTimer)
+		}
+	}
+
+	return nil
+}
+
+func boolToYesNo(value *bool, defaultYes bool) string {
+	return boolToString(value, defaultYes, "yes", "no")
+}
+
+func boolToString(value *bool, defaultPositive bool, positive string, negative string) string {
+	toString := func(value bool) string {
+		if value {
+			return positive
+		}
+		return negative
+	}
+
+	if value == nil {
+		return toString(defaultPositive)
+	}
+	return toString(*value)
+}
```

Comment:
`boolToString` is only called once - only from  `boolToYesNo`.

`defaultYes` is actually always `true`. But even if we do use it, its name is confusing, suggesting the the default is `true`.

We can simplify. Something like this:


```suggestion
func boolToYesNo(value *bool, defaultVal bool) string {
	val :=  ptr.Deref(value *bool, defaultVal)
	if val {
		return "yes"
	}
	
	return "no"
}
```

Or even:
```suggestion
func boolToYesNo(value *bool) string {
	if ptr.Deref(value *bool, true)
		return "yes"
	}
	
	return "no"
}